### PR TITLE
server: fix data race

### PR DESCRIPTION
**Pull request by nolouch**
PTAL @siddontang @disksing

### CODE REVIEW

**Comment by disksing** [comment]
LGTM

File: server/replication.go [hunk 1]
```diff
@@ -13,12 +13,16 @@
...
+import (
+	"math"
+	"sync"
+)
 
 const replicaBaseScore = 100
 
 // Replication provides some help to do replication.
 type Replication struct {
+	mu  sync.RWMutex
```
Comment (siddontang) [inline server/replication.go]: the mu is only for config, use muConfig or a struct like:

```
cfg struct {
    RwMutex
    *RepliactionConfig
}
```

Comment (siddontang) [inline server/replication.go]: you can also use atomic set/get directly.

File: _vendor/vendor/vendor [hunk 2]
```diff
@@ -0,0 +1 @@
+_vendor/vendor
```
Comment (siddontang) [inline _vendor/vendor/vendor]: what is this?

File: server/replication.go [hunk 3]
```diff
@@ -15,43 +15,56 @@ package server
...
+	"sync/atomic"
 )
 
 const replicaBaseScore = 100
 
 // Replication provides some help to do replication.
 type Replication struct {
-	mu  sync.RWMutex
-	cfg *ReplicationConfig
+	v atomic.Value
```
Comment (siddontang) [inline server/replication.go]: replicateCfg

File: _vendor/vendor/vendor (hunk 2 above)
Comment (nolouch) [inline _vendor/vendor/vendor]: it was removed

**Comment by siddontang** [comment]
LGTM
PTAL @disksing

**Comment by disksing** [comment]
LGTM

**Pull request merged**